models/aether_cube_sphere/model_mod.f90

```diff
@@ -0,0 +1,868 @@
+                              init_time => fail_init_time,                             &
+                              init_conditions => fail_init_conditions,                 &
+                              convert_vertical_obs, convert_vertical_state, adv_1step
+
+implicit none
+private
+
+! routines required by DART code - will be called from filter and other DART executables. 
+public :: get_model_size,                      &
+          get_state_meta_data,                 &
+          model_interpolate,                   &
+          end_model,                           &
+          static_init_model,                   &
+          nc_write_model_atts,                 &
+          get_close_obs,                       &
+          get_close_state,                     &
+          pert_model_copies,                   &
+          convert_vertical_obs,                &
+          convert_vertical_state,              &
+          read_model_time,                     &
+          adv_1step,                           &
+          init_time,                           &
+          init_conditions,                     &
+          shortest_time_between_assimilations, &
+          write_model_time
+
+! Routine for comprehensive test of interpolation
+public :: test_grid_box
+
+! version controlled file description for error handling, do not edit
+character(len=*), parameter :: source   = "$URL$"
+character(len=*), parameter :: revision = "$Revision$"
+character(len=*), parameter :: revdate  = "$Date$"
+
+! Error codes
+integer, parameter :: INVALID_VERT_COORD_ERROR_CODE   = 15
+integer, parameter :: INVALID_ALTITUDE_VAL_ERROR_CODE = 17
+integer, parameter :: INVALID_MODEL_LEVEL_ERROR_CODE  = 18
+integer, parameter :: UNKNOWN_OBS_QTY_ERROR_CODE      = 20
+
+! Error message strings                
+character(len=512) :: string1
+character(len=512) :: string2
+
+integer  :: iunit, io
+
+logical :: module_initialized = .false.
+integer :: dom_id ! used to access the state structure
+type(time_type) :: assimilation_time_step
+
+! Geometry variables that are used throughout the module; read from a template file
+integer               :: np                ! Number of grid rows across a face
+real(r8)              :: del, half_del     ! Grid row spacing and half of that
+integer               :: ncenter_altitudes ! The number of altitudes and the altitudes
+real(r8), allocatable :: center_altitude(:)
+
+! Current model time needed for computing location for scalar F10.7
+type(time_type) :: state_time
+
+! Horizontal column dimension rather than being direct functions of latitude and longitude.
+integer               :: no_third_dimension = -99
+
+type :: var_type
+    integer :: count
+    character(len=64), allocatable :: names(:)
+    integer,           allocatable :: qtys(:)
+    real(r8),          allocatable :: clamp_values(:, :)
+    logical,           allocatable :: updates(:)
+end type var_type
```

Comment:
This is implemented as a utility in https://github.com/NCAR/DART/releases/tag/v11.10.3
No need to re-implement this is a model_mod